recipes/nng/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+from glob import glob
+import os
+
+
+class NngConan(ConanFile):
+    name = "nng"
+    description = "nanomsg-next-generation: light-weight brokerless messaging"
+    url = "https://github.com/conan-io/conan-center-index"
+    exports_sources = ["CMakeLists.txt", "src/*.cmake", "src/*.txt"]
+    homepage = "https://github.com/nanomsg/nng"
+    license = "MIT"
+    topics = ("nanomsg", "communication", "messaging", "protocols")
+    generators = "cmake"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "tests": [True, False],
+        "nngcat": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "tests": False,
+        "nngcat": False
+    }
+
+    _source_subfolder = "source_subfolder"
+    _cmake = None
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+        if self.settings.compiler == "Visual Studio" and \
+                tools.Version(self.settings.compiler.version) < 14:
+            raise ConanInvalidConfiguration("MSVC < 14 is not supported")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        self._cmake = CMake(self)
+        self._cmake.definitions["NNG_TESTS"] = self.options.tests
+        self._cmake.definitions["NNG_NNGCAT"] = self.options.nngcat
+        self._cmake.configure(source_folder=self._source_subfolder)
+
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
```

Comment:
When enabling the tests, the resulting package will not be different because the tests are not installed.
So remove the option when calculating the package_id.
An alternative would be to remove the `tests` option altogether and disable building it unconditionally.
```suggestion
        cmake.build()

    def package_id(self):
        del self.info.options.tests

    def package(self):
```